source/server/filter_chain_manager_impl.h

```diff
@@ -71,8 +77,37 @@ class FilterChainFactoryContextImpl : public Configuration::FilterChainFactoryCo
   Configuration::TransportSocketFactoryContext& getTransportSocketFactoryContext() const override;
   Stats::Scope& listenerScope() override;
 
+  void setDraining() { is_draining_.store(true, std::memory_order_release); }
```

Comment:
I think we can avoid the explicit memory order complexity for this. Let's just go with `is_draining_ = true;`

Reply:
I think it's a straightforward publish/subscribe semantic so that I choose acquire/release
But I agree it's not a critical code path.